### PR TITLE
Changes documentation examples to async/await

### PR DESCRIPTION
**Pull request by NeedleInAJayStack**
This just adds a bit of documentation on how to use async functions in the resolvers.

### CODE REVIEW

**Review by paulofaria — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -200,13 +200,11 @@ defer {
     try? group.syncShutdownGracefully()
 }
 
-api.execute(
+try await api.execute(
```
Comment (adam-fowler) [inline README.md:203]: Could you replicate the functionality ie
```swift
let result = try await api.execute(...
print(result)
```
Or does that not make sense

**Review by adam-fowler — COMMENTED** [review]
Do we need the NIO resolvers section?

File: README.md (hunk 1 above)
Comment (NeedleInAJayStack) [reply README.md]: Whoops, didn't mean to remove the print. Thanks for catching this!

**Comment by NeedleInAJayStack** [comment]
> Do we need the NIO resolvers section?

I suppose we don't, but I feel like it's nice to have a usage example if it's functionality that we support.

**Comment by adam-fowler** [comment]
Ok fair enough

**Review by adam-fowler — APPROVED** [review]

**Pull request merged**